### DISCUSSION TITLE
Enable major updates only for some base branches

### DISCUSSION THREAD

**Discussion by astellingwerf** [renovatebot/renovate#16723] [Request Help] [answered]
I guess I must be missing something obvious here, but I can't get this to work. Can I enable major updates only for some base branches? My guess would be:

```
  "major": {
    "enabled": false
  },
  "packageRules": [
    {
      "matchBaseBranches": [
        "main"
      ],
      "major": {
        "enabled": true
      }
    }
  ]
```

But that gives an error: `Each packageRule must contain at least one non-match* or non-exclude* field.`.

Any suggestions?

**Comment by viceice** [comment, marked as answer]
use match update types instead of nested object

**Reply by astellingwerf** [reply]
But I'm not looking to apply a setting to major updates, I'm hoping to enable major updates.

**Reply by rarkins** [reply]
There is some massage code we use to flatten nesting within packageRules, which seems to be malfunctioning, and should be treated/fixed as a bug. But to avoid the problem in future, don't nest objects within packageRules (e.g. use `matchUpateTypes` and not `major`)

**Reply by astellingwerf** [reply]
Ah, wait, you mean?:
```
  "major": {
    "enabled": false
  },
  "packageRules": [
    {
      "matchBaseBranches": [
        "master"
      ],
      "matchUpdateTypes": [
        "major"
      ],
      "enabled": true
    }
  ]
```

**Reply by viceice** [reply]
yes

**Reply by viceice** [reply]
@rarkins we should probably add a warning for nested package rule objects?

**Reply by HonkingGoose** [reply]
> @rarkins we should probably add a warning for nested package rule objects?

I created a PR to put a warning in the docs:

- https://github.com/renovatebot/renovate/pull/16755

But maybe more work is needed to:

- Update Renovate config validator code so it warns about nested package rule objects?
- Let Renovate raise a config warning issue for nested package rules objects?

I'll let you maintainers decide what to do. 😉

**Reply by HonkingGoose** [reply]
> I'll let you maintainers decide what to do. 

Looks like we made a decision already. See: 

- #15464
